sdk/appcenter-data/src/main/java/com/microsoft/appcenter/data/Data.java

```diff
@@ -595,6 +596,17 @@ public void onCallFailed(Exception e) {
                     @Override
                     public void onCallSucceeded(String payload, Map<String, String> headers) {
                         Page<T> page = Utils.parseDocuments(payload, documentType);
+                        String tableName = Utils.getTableName(tokenResult);
+                        if (page.getItems() != null) {
+                            Iterator<DocumentWrapper<T>> documentWrapperIterator = page.getItems().iterator();
+                            while (documentWrapperIterator.hasNext()) {
```

Comment:
code analysis warning here

Reply:
It is recommended to commit with Android Studio commit dialog and check the option to check for code analysis (and almost all other options as well except todo which can stay in a PR until another task).

Reply:
If you are referring to converting while to for each, then we did this on purpose because this iterator gets called twice.  If we convert this to for each the iterator's cursor gets moved and the second iteration through pending documents won't work.

Reply:
You need to suppress the warnings then and comment